RFS/src/main/java/com/rfs/tracing/RootRfsContext.java

```diff
@@ -0,0 +1,51 @@
+package com.rfs.tracing;
+
+import io.opentelemetry.api.OpenTelemetry;
+import org.opensearch.migrations.tracing.IContextTracker;
+import org.opensearch.migrations.tracing.RootOtelContext;
+
+public class RootRfsContext extends RootOtelContext {
+    public static final String SCOPE_NAME = "rfs";
+    public final RfsContexts.CreateSnapshotContext.MetricInstruments snapshotInstruments;
+    public final RfsContexts.ClusterMetadataContext.MetricInstruments metadataMetrics;
+    public final RfsContexts.MigrateTemplateContext.MetricInstruments indexTemplateInstruments;
+    public final RfsContexts.GenericRequestContext.MetricInstruments genericRequestInstruments;
+    public final RfsContexts.CheckedIdempotentPutRequestContext.MetricInstruments getTwoStepIdempotentRequestInstruments;
+    public final RfsContexts.CreateIndexContext.MetricInstruments createIndexInstruments;
+    public final RfsContexts.DocumentReindexContext.MetricInstruments documentReindexInstruments;
+    public final RfsContexts.WorkingStateContext.MetricInstruments workingStateUpdateInstruments;
```

Comment:
1) Why are all of these made and accessible outside the root context, I'm not sure what the case for access is?
2) Shouldn't they be created and passed into `create***Context(...)`?  Note if this is about singular creation, lets use a lazy supplier